dotenv-linter/src/checks.rs

```diff
@@ -81,6 +88,11 @@ pub fn run(lines: &[LineEntry], skip_checks: &[LintKind]) -> Vec<Warning> {
         }
     }
 
+    for ch in &mut checks {
+        let end_warns = ch.end();
+        warnings.extend(end_warns);
+    }
```

Comment:
```suggestion
    if disabled_checks.contains(&LintKind::SchemaViolation) {
        return warnings;
    }

    if let Some(schema) = schema {
        let keys = lines.iter().filter_map(|l| l.get_key()).collect::<Vec<&str>>();
        let required_warnings = schema_violation::check_required_keys(schema, &keys);
        warnings.extend(required_warnings);
    }
```
It seems to me that it's more obvious what's going on.

Reply:
the 'end' call was intended as a generic call for any checkers that may want to do a final pass of some kind

This now elevates the schema checker to a special case